src/tcgwars/logic/impl/gen4/GreatEncounters.groovy

```diff
@@ -1836,10 +1835,17 @@ public enum GreatEncounters implements LogicCardInfo {
                   eff.unregister()
                   bg.em().run(new ChangeImplementation(top, toolCard))
                 }
+                onDisable {
+                  eff.unregister()
+                }
               }
               bg.em().run(new ChangeImplementation(toolCard, top))
               toolCard.player = top.player
-              attachPokemonTool(toolCard, my.all.select("Attach to?"))
+              def pcs = my.all.findAll {it!=self && canAttachPokemonTool(it)}.select("Attach to?")
+              removeFromPlay(self, [top] as CardList)
+              bg.em().run(new ChangeImplementation(toolCard, top))
+              attachPokemonTool(toolCard, pcs)
+              removePCS(self)
```

Comment:
removePCS() -> attachPokemonTool() seems simpler to me. is there any reason for picking this solution?

Reply:
I've made changes to the main-engine side of these changes that may be relevant. Basically, I am/want to use MoveCard instead of manually adding to/removing from card lists, because I've not seen that be anything but trouble while trying to implement certain cards.

Reply:
If using MoveCard in AttachPokemonTool, the card still needs to exist to be moved. Removing the PCS will remove the PCS entirely so the card will no longer exist on the field to be moved if removePCS is done first.